src/snapshot/tree.rs

```diff
@@ -161,11 +179,35 @@ impl RojoTree {
         self.metadata_map.get(&id)
     }
 
+    pub fn get_specified_id(&self, specified: &RojoRef) -> Option<Ref> {
+        match self.specified_id_to_refs.get(specified)[..] {
+            [referent] => Some(referent),
+            _ => None,
+        }
+    }
```

Comment:
```suggestion
    /// Get the backing Ref of the given RojoRef. If the RojoRef maps to exactly 
    /// one Ref, this method returns Some. Otherwise, it returns None.
    pub fn get_specified_id(&self, specified: &RojoRef) -> Option<Ref> {
        match self.specified_id_to_refs.get(specified)[..] {
            [referent] => Some(referent),
            _ => None,
        }
    }
```